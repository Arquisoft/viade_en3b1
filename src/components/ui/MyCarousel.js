import React, { Component } from 'react';
import AwesomeSlider from 'react-awesome-slider';
import withAutoplay from 'react-awesome-slider/dist/autoplay';
import 'react-awesome-slider/dist/styles.css';

const AutoplaySlider = withAutoplay(AwesomeSlider);

export class MyCarousel extends Component {

    constructor(props) {
        super(props);
        this.state = {
            photos: this.props.photos,
        }

        this.media = [];
        this.state.photos.forEach((p) => {
            this.media.push({ source: p.getContent() });
        });
    }

    render() {

        return (

            <AutoplaySlider
                play={true}
                cancelOnInteraction={false}
                interval={6000}
                media={this.media}
            />
        );
    }
}

export default MyCarousel;