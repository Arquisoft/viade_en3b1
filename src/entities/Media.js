class Media{
    constructor(url){
        this.url=url;
    }
}

export default Media;