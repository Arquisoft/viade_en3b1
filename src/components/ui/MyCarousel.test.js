import Media from "../../entities/Media";
import MyCarousel from "./MyCarousel";
import photo from '../../assets/img/fondo/city_1.jpg';
import photo2 from '../../assets/img/fondo/city_2.jpg';

var media = [new Media(photo), new Media(photo2)];

test('Define MyCarousel', () => {
    expect(MyCarousel).toBeDefined();
});

test('Renders correctly', () => {
    const wrapper = render(<MyCarousel photos={media} />);
    expect(wrapper).toMatchSnapshot();
});