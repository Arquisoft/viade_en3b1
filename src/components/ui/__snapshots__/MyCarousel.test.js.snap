// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  class="awssld awssld--organic-arrows"
>
  <div
    class="awssld__wrapper"
  >
    <div
      class="awssld__container"
    >
      <div
        class="awssld__box"
      >
        <div
          class="awssld__content"
        >
          <img
            src="city_1.jpg"
          />
        </div>
      </div>
      <div
        class="awssld__box"
      />
    </div>
    <div
      class="awssld__controls awssld__controls--hidden"
    >
      <button
        aria-label="next"
        class="awssld__next"
      >
        <span
          class="awssld__controls__arrow-right"
        />
      </button>
      <button
        aria-label="previous"
        class="awssld__prev"
      >
        <span
          class="awssld__controls__arrow-left"
        />
      </button>
    </div>
  </div>
  <nav
    class="awssld__bullets"
  >
    <button
      class="awssld__bullets--active"
      data-index="0"
    >
      0
    </button>
    <button
      data-index="1"
    >
      1
    </button>
  </nav>
</div>
`;
